### PR TITLE
fix(#8868): don't forward all headers for get requests

### PR DESCRIPTION
**Pull request by dianabarsan**
# Description

Node 19 adds keep-alive header to requests by default. This makes the `content-length` header very significant **and** important because it enables the server to know when one request has been fully received and another one is sent through the same connection. 
When authenticating a user, we take all headers from the original user's request and pass them to a GET /_session request, presumably so we don't have to filter out all the ways through which the user can authenticate (cookie, authorization etc). This meant that if the original request was a POST, and it had a `content-length` header, this header was also forwarded. This meant that when the connection would later be reused, the server (haproxy in this case) would truncate the previous's request `content-length` number of characters from the next request, believing the previous request was not finished. Obviously, this new truncated request was invalid and generated a 400 status code. 

This only seems to happen when accessing API directly or through the AWS load balancer, and never when running requests through nginx. 

To avoid next request truncation, I am no longer forwarding `content-length` headers in the session request. 

#8868

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# Compose URLs
If Build CI hasn't passed, these may 404:

* [Core](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8868-dont-forward-all-headers-for-get/docker-compose/cht-core.yml)
* [CouchDB Single](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8868-dont-forward-all-headers-for-get/docker-compose/cht-couchdb.yml)
* [CouchDB Cluster](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8868-dont-forward-all-headers-for-get/docker-compose/cht-couchdb-clustered.yml)

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by dianabarsan** [comment]
Added an e2e test to demonstrate fixed behavior, and an opposite e2e test here: https://github.com/medic/cht-core/compare/test-login?expand=1 that shows the 400 request: https://github.com/medic/cht-core/actions/runs/8274351579/job/22639815092#step:17:286

**Comment by dianabarsan** [comment]
Hi @latin-panda 

Would you mind reviewing this? Thanks a lot!

File: api/tests/mocha/auth.spec.js [hunk 1]
```diff
@@ -116,4 +126,79 @@ describe('Auth', () => {
...
+      }]]);
+    });
+
+    it('should clean content-length headers before forwarding', async () => {
+      sinon.stub(rpn, 'get').resolves({ userCtx: { name: 'theuser', roles: ['userrole'] }});
+
+      req.headers['content-length'] = 100;
+      req.headers['Content-Length'] = 22;
+      req.headers['Content-length'] = 44;
+      req.headers['content-Length'] = 82;
```
Comment (latin-panda) [inline api/tests/mocha/auth.spec.js:153]:
```suggestion
      req.headers['content-Length'] = 82;
      req.headers['CONTENT-LENGTH'] = 240;
```

**Review by latin-panda — APPROVED** [review]
LGTM!

Thanks for explaining in the PR's description box, I learned from it!

**Pull request merged**